### PR TITLE
Finishing entity installation and upgrade

### PR DESCRIPTION
**Pull request by Kaik**
| Q | A
| ------------- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | yes
| Deprecations? | no
| Tests pass? | no
| Fixed tickets | #75  
| Refs tickets | #73  
| License | MIT
| Doc PR |

### CODE REVIEW

File: Util/Messages.php [hunk 1]
```diff
@@ -33,8 +34,79 @@ public function __construct()
...
+        $total['archive']['count'] = $stored->count();
+        $total['archive']['limit'] = 50;     
+        
+        return $total;
+    }    
+    
+}
+
 
     /**
```
Comment (shefik) [inline Util/Messages.php]: PHP is throwing the following error here:

```
Warning: Unterminated comment starting line 111 in /modules/InterCom/Util/Messages.php on line 111
```

**Comment by shefik** [comment]
@Kaik Thanks for submitting this pull request. I tested your pull request with a new installation, and it does install correctly. However, please correct the PHP error mentioned above. Someone will need to test the upgrade process with existing data. Also, please review your changes to all of the templates. There are multiple display issues. For example, in the attached screenshot, you can see how elements are overlapping and breaking layout in the user inbox template. In other templates, there are alignment issues, particularly at tablet breakpoints, where form labels are mis-aligned.

inbox

**Comment by Kaik** [comment]
@shefik I am aware of template issues there is many more things to do. This part is only related to table layout, install and upgrade.

**Comment by Kaik** [comment]
I have removed comments. There is still many things to do, at this stage table structure is done, install is done and upgrade is almost done. It should upgrade properly but there are some side effects "deleted user and admin/system user" messages. Additionally module dependencies and listener is not touched, this will be added afterwards.

**Comment by shefik** [comment]
@Kaik I tested your latest code. It appears to work for me with a new installation of the module. It also works when I send myself a test message. You mentioned "side effects" with the upgrade process. Can you elaborate? Are you saying that when you attempt an upgrade, it deletes existing messages from the database unexpectedly? If so, are you planning to address this issue in this same pull request via additional commits, or are you planning a different pull request specially for upgrades, or are you requesting assistance to troubleshoot?

**Comment by Kaik** [comment]
New table structure, actually new "user fields - sender, recipient"(where user data is saved as doctrine entity)... its specification AFAIK to do not allow user id to be 0(or it is guest id etc anyway) so we have FALSE and all positive values aka user id's to use. For false I guess it will implicate deleted user and rest positives are user id's. Now for messages we need third case which is "system user id" it can be one of admin's... we need a way to indicate this system messages and those are side effects I'm talking about. Upgrade is do not delete anything it may mark messages "as deleted" so they may appear as deleted but there are still in db

This pull request is actually kind of milestone where I have defined new model and mostly upgraded to. Upgrade has one, known for me, issue all "system messages" end up as null which is deleted user so this is wrong. 
Additionally CRUD need to be checked and updated because things have changed.
I will do some clearing (new pr) hopefully next week and then next, need to do something first... Yes I hope some testing, any things you will find odd etc. please let me know.

**Comment by Kaik** [comment]
Ok @shefik you can check now I have made some more commits including mentioned view issues and saving/deleteting message problem.

**Comment by shefik** [comment]
@Kaik Something in the code of your pull request is somehow now breaking the Zikula core installation. Can you take a look? Try doing a clean installation of the core using Zikula 1.4.0 build 510, and also have your InterCom module in the /modules folder.

Refs. https://github.com/zikula/core/issues/2374

**Comment by shefik** [comment]
Appears to be working.

**Pull request merged**